internal/resources/providers/awslib/config.go

```diff
@@ -101,33 +117,46 @@ func InitializeAWSConfigCloudConnectors(ctx context.Context, cfg config.AwsConfi
 	return retConf, nil
 }
 
-// AWSConfigRoleChaining initializes an assume role provider and an credential cache for each step on the chain, using the previous one as client.
-func AWSConfigRoleChaining(initialConfig aws.Config, chain []AWSRoleChainingStep) *aws.Config {
-	var client *sts.Client
-	var assumeRoleProvider *stscreds.AssumeRoleProvider
-	var credentialsCache *aws.CredentialsCache
-	cnf := initialConfig
-
-	for _, c := range chain {
-		client = sts.NewFromConfig(cnf) // create client using the credentials from previous or initial step.
-
-		// create a assume role provider for the current chain part role.
-		assumeRoleProvider = stscreds.NewAssumeRoleProvider(
-			client,
-			c.RoleARN,
-			c.Options,
-		)
-		credentialsCache = aws.NewCredentialsCache(assumeRoleProvider)
-
-		cnf.Credentials = credentialsCache
+// NewAWSConfigOIDCChain creates an AWS config using OIDC/Web Identity token-based authentication with role chaining.
+// This function performs a two-step role assumption chain:
+//  1. Uses AssumeRoleWithWebIdentity to authenticate as the global role using a JWT token from the specified file
+//  2. Uses the global role credentials to assume the remote/target role with an external ID
+func NewAWSConfigOIDCChain(ctx context.Context, jwtFilePath string, cfg config.AwsConfig) (*aws.Config, error) {
+	const defaultDuration = 20 * time.Minute
+
+	// Load base AWS config
+	awsConfig, err := awsconfig.LoadDefaultConfig(ctx)
```

Comment:
We need to do something like that here 

https://github.com/elastic/beats/blob/8804508f8e4fcc9c844b3b0ee1f7533a3c1a62a0/x-pack/libbeat/common/aws/credentials.go#L60-L66